### PR TITLE
feature: added Page Speed Insights Data Studio connector which I used…

### PR DESCRIPTION
**Pull request by grantkemp**
… as a learning exercise to learn the api. Additionally I really found it useful to have Page Speed Insights in Google Analytics - so it was useful for our office to have the page speed score displayed for any key pages that we were making changes to - so the whole team could keep an eye on it and report it.

Since this is a Google owned product- I assume (and hope) others in Google/ the web performance / analytics community would like to add more features to it - such as :
1. breaking out mobile and desktop score into separate metrics
2. Listing the rules that are not complied with as a separate set of dimensions to be displayed in a table.
3. Other cool ideas to help improve speed of websites.

Any feedback - feel free to add issue or tweet me on [@ukdatageek](https://twitter.com/ukdatageek)

### CODE REVIEW

File: pagespeed-insights/README.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+- **3rd Party Key based authentication**  
+  This community connector requires the use of a third-party 'key' based authentication.  I have also done some rudimentary validating of the key eg checking for blank/ null values.
+- **Error handling and messaging**  
+  Example of using [error handling methods and providing useful error messages
+  to users](https://developers.google.com/datastudio/connector/error-handling).
+- **Using a 3rd party Api which returns the configuration as part of the Data Studio results**  
+  The connector uses a 3rd Party api to get the page speed score, but also returns the configuration which was set by the user as a custom dimension. 
+
+##Roadmap 
+Since this is a Google owned product- I assume (and hope) others in Google/ the web performance / analytics community would like to add more features to it - such as :
```
Comment (diminishedprime) [inline pagespeed-insights/README.md]: Could you clarify the text here? While the api is a Google owned product, the connectors in this repo are not. It's definitely good to call out a roadmap for the connector, but make sure that it's clear that Google is not supporting this connector (other than managing the deployments).

File: pagespeed-insights/main.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
+//NOTE: You can obtain a Google PAge Speed Insights API Key from here: https://developers.google.com/speed/docs/insights/v4/first-app
```
Comment (diminishedprime) [inline pagespeed-insights/main.js]: s/PAge/Page

**Comment by diminishedprime** [comment]
Thanks for the merge request! Just a few things

File: pagespeed-insights/main.js [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+    var urlparts = [
+        'https://www.googleapis.com/pagespeedonline/v4/runPagespeed?url=',
+        request.configParams.urlTotest,
+        '&strategy=mobile&key=',
+        key,
+        '&fields=ruleGroups'
+    ];
+    var url = urlparts.join('');
+    var response = UrlFetchApp.fetch(url);
```
[the remark this answers is not in the document]
Comment (grantkemp) [reply pagespeed-insights/main.js]: removed.

File: pagespeed-insights/main.js [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+    var userProperties = PropertiesService.getUserProperties();
+    var key = userProperties.getProperty('dscc.key');
+    var isValid = validateKey(key);
+    return isValid
+}
+
+function getConfig(request) {
+    var config = {
+        configParams: [
+            //TODO: Keep this in in case this is a simpler way to show apikey and give users ability to generate one
```
[the remark this answers is not in the document]
Comment (grantkemp) [reply pagespeed-insights/main.js]: Done

File: pagespeed-insights/README.md (hunk 1 above)
Comment (grantkemp) [reply pagespeed-insights/README.md]: I added more text to clarify that it was not Google Supported.

File: pagespeed-insights/main.js (hunk 2 above)
Comment (grantkemp) [reply pagespeed-insights/main.js]: sorted!

**Comment by grantkemp** [comment]
Thanks for merging @diminishedprime - I will keep an eye out for issues.

**Pull request merged**